pkg/apis/sources/v1alpha1/ping_lifecycle.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"knative.dev/pkg/apis"
+
+	"knative.dev/eventing/pkg/apis/duck"
+)
+
+const (
+	// PingSourceConditionReady has status True when the PingSource is ready to send events.
+	PingSourceConditionReady = apis.ConditionReady
+
+	// PingSourceConditionValidSchedule has status True when the PingSource has been configured with a valid schedule.
+	PingSourceConditionValidSchedule apis.ConditionType = "ValidSchedule"
+
+	// PingSourceConditionSinkProvided has status True when the PingSource has been configured with a sink target.
+	PingSourceConditionSinkProvided apis.ConditionType = "SinkProvided"
+
+	// PingSourceConditionDeployed has status True when the PingSource has had it's receive adapter deployment created.
+	PingSourceConditionDeployed apis.ConditionType = "Deployed"
+
+	// PingSourceConditionEventTypeProvided has status True when the PingSource has been configured with its event type.
+	PingSourceConditionEventTypeProvided apis.ConditionType = "EventTypeProvided"
+
+	// PingSourceConditionResources is True when the resources listed for the PingSource have been properly
+	// parsed and match specified syntax for resource quantities
+	PingSourceConditionResources apis.ConditionType = "ResourcesCorrect"
+)
+
+var PingSourceCondSet = apis.NewLivingConditionSet(
+	PingSourceConditionValidSchedule,
+	PingSourceConditionSinkProvided,
+	PingSourceConditionDeployed)
+
+// GetCondition returns the condition currently associated with the given type, or nil.
+func (s *PingSourceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
```

Comment:
nit: while here, maybe add the GetTopLevelCondition() method? Later is fine too :)

Reply:
added.